Minesweeper.java

```diff
@@ -0,0 +1,99 @@
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionListener;
+import java.util.*;
+import java.awt.event.*;
+
+public class Minesweeper {
+
+    // The size in pixels for the frame.
+    private static final int SIZE = 500;
+    private JFrame  frame;
+
+    private final ActionListener actionListener = actionEvent -> {
+        Object source = actionEvent.getSource();
+    };
+
+    private class Cell extends JButton {
+        private final int row;
+        private final int col;
+        private int value;
+
+        Cell(final int row, final int col,
+             final ActionListener actionListener) {
+            this.row = row;
+            this.col = col;
+            addActionListener(actionListener);
+            setText("");
+        }
+
+        //TODO: For later implementations
+        /*int getValue() {
+            return value;
+        }
+
+        void setValue(int value) {
+            this.value = value;
+        }
+
+        boolean isAMine() {
+            return value == MINE;
+        }*/
+
+        @Override
+        public boolean equals(Object obj) {
+            if (this == obj) return true;
+            if (obj == null || getClass() != obj.getClass())
+                return false;
+            Cell cell = (Cell) obj;
+            return row == cell.row &&
+                   col == cell.col;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(row, col);
+        }
+    }
+    private class Grid {
+        private int gridSize;
+        private Cell[][] cells;
+
+        public Grid(int gridSize){
+        
+            this.gridSize = gridSize;
+            this.cells = new Cell[gridSize][gridSize];
+
+            frame = new JFrame("Minesweeper");
```

Comment:
do we want the `Grid` class to handle the swing frame? what about when we will need a frame without having a grid, like with the initial menu screen? Note that the way it's implemented here means that the frame will only show up once the grid is initialized.

Perhaps we should have all of the frame-handling inside of a dedicated class for that, and that class could also switch between game state and start-menu state.

Reply:
yeah i tried separating them but i dont see why thats an issue, when we need another frame we will create/use one in the adequate class/method

Reply:
> yeah i tried separating them but i dont see why thats an issue, when we need another frame we will create/use one in the adequate class/method

im not sure but wouldn't that make it so the window closes and reopens every time the game state switches from menu screen to game and vice versa?

Reply:
well the game frame would close but the menu one would open, the same way you can picture it with different forms in C# WF

Reply:
i feel like it would be a better experience if the frame remained consistent the whole time you're playing, rather than having a window close and open each time you go back to the menu

Reply:
however i do recognize that this is outside of the scope of this pull request, so once this gets merged we should make a separate issue addressing this